satellite/metainfo/metainfo.go

```diff
@@ -467,9 +472,37 @@ func (endpoint *Endpoint) filterValidPieces(ctx context.Context, pointer *pb.Poi
 			// 	s.logger.Warn("unable to verify piece hash: %v", zap.Error(err))
 			// }
 
+			err = endpoint.validatePieceHash(ctx, piece, limits)
+			if err != nil {
+				// TODO should this be logged also to uplink somehow ?
+				endpoint.log.Sugar().Warn(err)
+				continue
+			}
+
+			// TODO maybe minimal PieceSize should be bigger
+			if piece.Hash.PieceSize <= 0 || (lastPieceSize > 0 && lastPieceSize != piece.Hash.PieceSize) {
+				allSizesValid = false
+				break
+			}
+			lastPieceSize = piece.Hash.PieceSize
+
 			remotePieces = append(remotePieces, piece)
 		}
 
+		if allSizesValid {
+			redundancy, err := eestream.NewRedundancyStrategyFromProto(pointer.GetRemote().GetRedundancy())
+			if err != nil {
+				return Error.New("unable to verify piece size, invalid redundancy scheme values")
```

Comment:
Here we lose the original error and it may be harder to debug what was exactly wrong with the redundancy. Perhaps, it's better to just return the error.

Reply:
I'm wondering if it would be useful to have a kind of function that allows to wrap the error and provides an error.